api/v1beta1/foundationdbcluster_types.go

```diff
@@ -1361,11 +1364,17 @@ func (cluster *FoundationDBCluster) CheckReconciliation() (bool, error) {
 	cluster.Status.Generations = ClusterGenerationStatus{Reconciled: cluster.Status.Generations.Reconciled}
 
 	for _, processGroup := range cluster.Status.ProcessGroups {
-		if processGroup.Remove && !processGroup.Excluded {
-			cluster.Status.Generations.NeedsShrink = cluster.ObjectMeta.Generation
-			reconciled = false
-		} else if processGroup.Remove {
-			cluster.Status.Generations.HasPendingRemoval = cluster.ObjectMeta.Generation
+		if processGroup.Remove {
+			terminating := false
+			for _, condition := range processGroup.ProcessGroupConditions {
+				terminating = terminating || condition.ProcessGroupConditionType == ResourcesTerminating
+			}
```

Comment:
```suggestion
			if processGroup.GetConditionTime(ResourcesTerminating) != nil {
				terminating = true
			}
```

Reply:
With that change we could also merge this if statement with the statement below.